.cloudbuild.yaml

```diff
@@ -94,6 +94,18 @@ steps:
          '--build-arg', 'COMMIT_HASH=$COMMIT_SHA', '-f',
          '/workspace/backend/metadata_writer/Dockerfile', '/workspace']
   waitFor: ["-"]
+- id: 'buildExecutionCache'
+name: 'gcr.io/cloud-builders/docker'
+args: ['build', '-t', 'gcr.io/$PROJECT_ID/execution-cache:$COMMIT_SHA',
+        '--build-arg', 'COMMIT_HASH=$COMMIT_SHA', '-f',
+        '/workspace/backend/execution_cache/Dockerfile', '/workspace']
```

Comment:
why this is called execution_cache?

Reply:
IMO the cache level is for every step execution. We will generate cache-key based on execution template, persist output of this execution, and prevent execution if it's already cached.

Reply:
Execution is a MLMD only noun. For Argo it's called step and I think we should give it more generic name to detach it from the implementation.